cli/flox-test-utils/src/proptest.rs

```diff
@@ -0,0 +1,214 @@
+use std::collections::{BTreeMap, BTreeSet};
+
+use prop::option;
+use proptest::collection::{btree_map, btree_set as prop_btree_set, vec as prop_vec};
+use proptest::prelude::*;
+
+pub fn chrono_strat() -> impl Strategy<Value = chrono::DateTime<chrono::Utc>> {
+    use chrono::TimeZone;
+
+    let start = chrono::Utc.with_ymd_and_hms(1970, 1, 1, 0, 0, 0).unwrap();
+    let end = chrono::Utc.with_ymd_and_hms(2100, 1, 1, 0, 0, 0).unwrap();
+
+    (start.timestamp()..end.timestamp())
+        .prop_map(|timestamp| chrono::Utc.timestamp_opt(timestamp, 0).unwrap())
+}
+
+/// Produces strings that only contain alphanumeric characters.
+///
+/// This is handy when you want to generate valid TOML keys without worrying about quoting
+/// or escaping.
+pub fn alphanum_string(max_size: usize) -> impl Strategy<Value = String> {
+    let ranges = vec!['a'..='z', 'A'..='Z', '0'..='9'];
+    prop::collection::vec(
+        proptest::char::ranges(std::borrow::Cow::Owned(ranges)),
+        1..=max_size,
+    )
+    .prop_map(|v| v.into_iter().collect())
+}
+
+/// Produces strings that only contain alphanumeric characters.
+///
+/// This is handy when you want to generate valid TOML keys without worrying about quoting
+/// or escaping.
+pub fn alphanum_and_whitespace_string(max_size: usize) -> impl Strategy<Value = String> {
+    let ranges = vec![
+        'a'..='z',
+        'A'..='Z',
+        '0'..='9',
+        ' '..=' ',
+        char::from(9)..=char::from(10),  // tab and lf
+        char::from(13)..=char::from(13), // cr
+    ];
+    prop::collection::vec(
+        proptest::char::ranges(std::borrow::Cow::Owned(ranges)),
+        1..=max_size,
+    )
+    .prop_map(|v| v.into_iter().collect())
+}
+
+/// Produces `Option<String>` instances with limited sizes for performance
+/// reasons.
+pub fn optional_string(string_max_size: usize) -> impl Strategy<Value = Option<String>> {
+    option::of(alphanum_string(string_max_size))
+}
+
+/// Produces maps whose keys are strings that only contain alphanumeric characters.
+pub fn btree_map_strat<T: Arbitrary>(
```

Comment:
i think we have the three characters to spare to keep at leas tme from wondering what a "strat" is

or we drop the strat part entirely since that;s also not present in `optional_btree_map`, `btree_set`, ...

```suggestion
pub fn btree_map_strategy<T: Arbitrary>(
```

Reply:
Yeah we can afford the extra three characters. I explicitly didn't call it `btree_map` to avoid confusion between the `std::collections::btree_map` module and the `proptest::collection::btree_map` function.

Reply:
I thought it would be clear from context that `strat` was short for `strategy` and that's a common thing to see when using the `hypothesis` library in Python, but I understand that it may not be obvious so I'll change it.

Reply:
Fixed